### PR TITLE
Shared Items

### PR DESCRIPTION
**Pull request by marek-vysoky**
Adding BoxSharedItemsManager, adding BoxSharedItemsManagerTest

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @marek-vysoky has signed the CLA. Thanks for the pull request!

**Comment by marek-vysoky** [comment]
Renaming SharedItems to SharedItemsString, SharedItemsEndpointUri to SharedItemsUri, changing comment for service in BoxSharedItemsManager

File: Box.V2/Managers/BoxSharedItemsManager.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+using Box.V2.Extensions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Box.V2.Managers
+{
+
```
Comment (coolboy) [inline Box.V2/Managers/BoxSharedItemsManager.cs]: remove the empty line

File: Box.V2/Managers/BoxSharedItemsManager.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        public BoxSharedItemsManager(IBoxConfig config, IBoxService service, IBoxConverter converter, IAuthRepository auth, string asUser = null, bool? suppressNotifications = null)
+            : base(config, service, converter, auth, asUser, suppressNotifications) { }
+
+        /// <summary>
+        /// Shared items are any files or folders that are represented by a shared link. Shared items are different from other API resources in that a shared resource doesn’t necessarily have to be in the account of the user accessing it. The actual shared link itself is used along with a normal access token.
+        /// </summary>
+        /// <param name="sharedLink">The shared link for this item.</param>
+        /// <returns>A full file or folder object is returned if the shared link is valid and the user has access to it. An error may be returned if the link is invalid, if a password is required, or if the user does not have access to the file.</returns>
+        public async Task<BoxItem> SharedItemsAsync(string sharedLink)
```
Comment (coolboy) [inline Box.V2/Managers/BoxSharedItemsManager.cs]: do we want to support shared_link_password as well?

**Comment by marek-vysoky** [comment]
@coolboy I've added shared_link_password support

**Comment by cburnette** [comment]
@coolboy Need to review this.

**Pull request merged**